user/module/probe_dash.go

```diff
@@ -0,0 +1,311 @@
+		},
+		{
+			Name:  "target_uid",
+			Value: uint64(b.conf.GetUid()),
+			//FailOnMissing: true,
+		},
+		{
+			Name:  "target_errno",
+			Value: uint64(b.Module.conf.(*config.DashConfig).ErrNo),
+		},
+	}
+
+	if b.conf.GetPid() <= 0 {
+		b.logger.Info().Msg("target all process.")
+	} else {
+		b.logger.Info().Uint64("target PID", b.conf.GetPid()).Send()
+	}
+
+	if b.conf.GetUid() <= 0 {
+		b.logger.Info().Msg("target all users.")
+	} else {
+		b.logger.Info().Uint64("target UID", b.conf.GetUid()).Send()
+	}
+
+	return editor
+}
+
+func calReadFuncAddressByGDB(elfPath string) (uint64, error) {
+
+	cmd := fmt.Sprintf("%s %s --batch -ex 'printf \"%%p\", %s'", elfPath, "/bin/dash", "read")
+	out, err := exec.Command("/bin/sh", "-c", cmd).Output()
+	if err != nil {
+		return 0, err
+	}
+	hexStr := strings.TrimSpace(string(out))
+	output := strings.TrimPrefix(hexStr, "0x")
+	ui64, err := strconv.ParseUint(output, 16, 64)
+	if err != nil {
+		return 0, err
+	}
+	return ui64, nil
+}
+
+func calReadFuncAddressByObjdump(elfPath string) (uint64, error) {
+	//0000000000005800 <read@plt>:
+	cmd := fmt.Sprintf("%s -d /bin/dash | grep read@plt | head -n 1", elfPath)
+	out, err := exec.Command("/bin/sh", "-c", cmd).Output()
+	if err != nil {
+		return 0, err
+	}
+	outputStr := strings.TrimSpace(string(out))
+	split := strings.Split(outputStr, " ")
+	if len(split) < 1 {
+		return 0, fmt.Errorf("objdump not expected result:%s", outputStr)
+	}
+	output := strings.TrimPrefix(split[0], "0")
+	ui64, err := strconv.ParseUint(output, 16, 64)
+	if err != nil {
+		return 0, err
+	}
+	return ui64, nil
+}
+func (b *MDashProbe) calReadFuncAddress() (uint64, error) {
+
+	var findElfBinErr, runCommandErr error
+	var elfPath string
+	var addr uint64
+	if elfPath, findElfBinErr = exec.LookPath("gdb"); findElfBinErr == nil {
+		if addr, runCommandErr = calReadFuncAddressByGDB(elfPath); runCommandErr == nil {
+			return addr, nil
+		}
+	}
+	//zap.S().Warnf("calReadFuncAddressByGDB failed,findElfBinErr:%v,runCommandErr:%v", findElfBinErr, runCommandErr)
+	if elfPath, findElfBinErr = exec.LookPath("objdump"); findElfBinErr == nil {
+		if addr, runCommandErr = calReadFuncAddressByObjdump(elfPath); runCommandErr == nil {
+			return addr, nil
+		}
+	}
+	//zap.S().Warnf("calReadFuncAddressByObjdump failed,findElfBinErr:%v,runCommandErr:%v", findElfBinErr, runCommandErr)
+
+	return 0, errors.New("use gdb and objdump to calReadFuncAddress failed")
+}
+
+func (b *MDashProbe) setupManagers() {
+	var binaryPath string
+	switch b.conf.(*config.DashConfig).ElfType {
+	case config.ElfTypeBin:
+		binaryPath = b.conf.(*config.DashConfig).Dashpath
+	case config.ElfTypeSo:
+		binaryPath = b.conf.(*config.DashConfig).Readline
+	default:
+		binaryPath = "/bin/dash"
+	}
+
+	var readlineFuncName string // 将默认hook函数改为readline_internal_teardown说明：https://github.com/gojue/ecapture/pull/479
+	readlineFuncName = b.conf.(*config.DashConfig).ReadlineFuncName
+	addr, err := b.calReadFuncAddress()
```

Comment:
`addr` should be a configuration item, passed in by the `config.DashConfig` structure, that is to say, the function of finding the offset address should be placed in the `config` package.